client/trino-cli/src/main/java/io/trino/cli/Trino.java

```diff
@@ -34,12 +44,43 @@ public static void main(String[] args)
 
     public static CommandLine createCommandLine(Object command)
     {
-        return new CommandLine(command)
+        CommandLine commandLine = new CommandLine(command)
                 .registerConverter(ClientResourceEstimate.class, ClientResourceEstimate::new)
                 .registerConverter(ClientSessionProperty.class, ClientSessionProperty::new)
                 .registerConverter(ClientExtraCredential.class, ClientExtraCredential::new)
                 .registerConverter(HostAndPort.class, HostAndPort::fromString)
                 .registerConverter(Duration.class, Duration::valueOf);
+
+        getConfigFile().ifPresent(file -> commandLine.setDefaultValueProvider(new CommandLine.PropertiesDefaultProvider(file)));
+
+        return commandLine;
+    }
+
+    private static Optional<File> getConfigFile()
+    {
+        return getConfigSearchPaths()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(Paths::get)
+                .filter(Files::exists)
+                .findFirst()
+                .map(Path::toFile);
+    }
+
+    private static Optional<String> resolveConfigPath(String root, String file)
+    {
+        return Optional.ofNullable(emptyToNull(root))
+                .map(Paths::get)
+                .filter(Files::exists)
+                .map(path -> path.resolve(file).toString());
+    }
+
+    private static Stream<Optional<String>> getConfigSearchPaths()
+    {
+        return Stream.of(
+                Optional.ofNullable(emptyToNull(getenv("TRINO_CONFIG"))),
+                resolveConfigPath(USER_HOME.value(), ".trino_config"),
+                resolveConfigPath(getenv("XDG_CONFIG_HOME"), "/trino/config"));
```

Comment:
handle case when `XDG_CONFIG_HOME` is not defined

Reply:
it is handled :) see resolveConfigPath method. it checks whether root exists first (and is not empty)

Reply:
👍

Reply:
```
private static Optional<String> resolveConfigPath(String root, String file)
    {
        return Optional.ofNullable(emptyToNull(root))
                .map(Paths::get)
                .filter(Files::exists)
                .map(path -> path.resolve(file).toString());
    }
```